core/src/PrognosticData.cpp

```diff
@@ -51,6 +62,10 @@ void PrognosticData::configure()
     tryConfigure(pDynamics);
 
     tryConfigure(iceGrowth);
+
+    if (getConfiguration(keyMap.at(CHECKFIELDS_KEY), checkFieldsDefault)) {
+        setFieldsToCheck(getConfiguration(keyMap.at(FIELDNAMES_KEY), fieldNamesDefault), pfx);
+    }
```

Comment:
Put the common code and configuration keys into the `CheckingModelComponent` class.

Reply:
I don't know how to do that.

``checkFieldsDefault`` and ``fieldNamesDefault`` are unique to the child classes ``PrognosticData``, ``ERA5Atmosphere``, or ``TOPAZOcean``, or other unimplemented uses of ``CheckingModelComponent``. The default field list should depend on the child class, and whether to check or not should be true for ``PrognosticData`` and false for everyone else.

``keyMap`` is also unique to the child class because the string can have different prefixes, e.g. ``ERA5Atmosphere.check_fields`` vs. ``TOPAZOcean.check_fields``.

So, I agree that there's too much repeated code, but I don't see how to reduce it meaningfully.